ui/appui-react/src/appui-react/imodel-components/models-tree/ModelsTree.tsx

```diff
@@ -206,9 +204,13 @@ function useVisibilityHandler(
   filteredDataProvider?: IFilteredPresentationTreeDataProvider,
   hierarchyAutoUpdateEnabled?: boolean,
 ) {
+  const iModel = activeView?.iModel;
+  const subjectModelIdsCache = React.useMemo(() => iModel ? new SubjectModelIdsCache(iModel) : undefined, [iModel]);
```

Comment:
Can we use iModelConnection from ModelsTree props to avoid checks for `undefined`?

Reply:
done